scrapy_poet/page_input_providers.py

```diff
@@ -198,3 +200,26 @@ def deserialize(self, data: Any) -> Sequence[Any]:
             )
             for response_data in data
         ]
+
+
+class HttpClientProvider(PageObjectInputProvider):
+    """This class provides ``web_poet.requests.HttpClient`` instances."""
+    provided_classes = {HttpClient}
+
+    def __call__(self, to_provide: Set[Callable], crawler: Crawler):
+        """Creates an ``web_poet.requests.HttpClient`` instance using Scrapy's
+        downloader.
+        """
+        backend = create_scrapy_backend(crawler.engine.download)
+        return [HttpClient(request_downloader=backend)]
+
+
+class MetaProvider(PageObjectInputProvider):
+    """This class provides ``web_poet.page_inputs.Meta`` instances."""
+    provided_classes = {Meta}
+
+    def __call__(self, to_provide: Set[Callable], request: Request):
+        """Creates a ``web_poet.requests.Meta`` instance based on the data found
+        from the ``meta["po_meta"]`` field of a ``scrapy.http.Response`` instance.
+        """
+        return [Meta(**request.meta.get("po_meta", {}))]
```

Comment:
Would this work? I wonder if some of the keys may not be valid argument names.

```suggestion
        return [Meta(request.meta.get("po_meta", {}))]
```

Reply:
TIL

```python
>>> dict(1=2)
  File "<stdin>", line 1
SyntaxError: keyword can't be an expression
>>> dict(**{"1": "2"})
{'1': '2'}
```

So I guess we could leave the `**` there, but it is probably better to remove it unless there is a good reason to keep it.